pkg/apis/pipeline/v1alpha1/artifact_bucket.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// BucketConfigName is the name of the configmap containing all
+	// customizations for the storage bucket.
+	BucketConfigName = "config-artifact-bucket"
+
+	// BucketLocationKey is the name of the configmap entry that specifies
+	// loction of the bucket.
+	BucketLocationKey = "location"
+
+	// BucketServiceAccountSecretName is the name of the configmap entry that specifies
+	// the name of the secret that will provide the servie account with bucket access.
+	// This secret must  have a key called serviceaccount that will have a value with
+	// the service account with access to the bucket
+	BucketServiceAccountSecretName = "bucket.service.account.secret.name"
+
+	// BucketServiceAccountSecretKey is the name of the configmap entry that specifies
+	// the secret key that will have a value with the service account json with access
+	// to the bucket
+	BucketServiceAccountSecretKey = "bucket.service.account.secret.key"
+)
+
+var (
+	secretVolumeMountPath = "/var/bucketsecret"
+)
+
+// ArtifactBucket contains the Storage bucket configuration defined in the
+// Bucket config map.
+type ArtifactBucket struct {
+	Name     string
+	Location string
+	Secrets  []SecretParam
+}
+
+// IsPVC indicates if the temporary storage used for artifacts in a pipelinerun is a PVC
+func (b *ArtifactBucket) IsPVC() bool {
```

Comment:
Would it be a bit safer to make this return a constant string rather than a boolean here, in case we end up with a third option later?

Reply:
will change it to a type with a const string value